tests/orm/data/test_code.py

```diff
@@ -0,0 +1,51 @@
+"""Tests for the `Code` class."""
+# pylint: disable=redefined-outer-name
+import pytest
+
+from aiida.orm import Code
+
+
+@pytest.fixture
+def create_codes(tmpdir, aiida_localhost):
+    """Create a local and remote `Code` instance."""
+    import os
+
+    filename = 'add.sh'
+
+    # Cast `tmpdir` to str as it is a Path object and Python 3.5 does not support Path objects for `join` yet
+    dirpath = str(tmpdir)
+    filepath = os.path.join(dirpath, filename)
```

Comment:
You could also cast to ``str`` here, and use the glorious ``/`` operator for joining:

```python
filepath = str(dirpath / filename)
```

Reply:
nice! It is clear I have not yet worked with `Path`s at all. Was going to merge because I am sick of waiting for the builds of this PR, but this is too nice to pass on

Reply:
But, then I don't even need the cast right? Or does `open` also not support `Path` in 3.5?

Reply:
You still need the cast, `pathlib` support across the board only came in 3.6.

Reply:
I guess we're back to waiting for another build 😅